modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/ArrayRowHandler.java

```diff
@@ -72,6 +73,34 @@ private ArrayRowHandler() {}
 
                 return fields;
             }
+
+            /**
+             * Wraps a column collection to {@code Object[]} if requred using the knowladge of expected column number.
+             *
+             * @param rowContainer Row values holder. Must be an {@code Iteratable} or an {@code Object[]}.
+             */
+            @Override public Object[] createByRowContainer(Object rowContainer) {
+                assert rowContainer instanceof Iterable || rowContainer.getClass() == Object[].class;
```

Comment:
1. Here should be correct user exception, not assert. Assert - for cases that can never happen. Here we depend on user input and everything is possible.
2. `Iterable` is to strong, let's use at least `Collection`, In real cases, I think, it will always be `List`.

Reply:
Ok, let's use a collection. I don't think we should use a real code like `if(...) throw ...` at each row being processed. Too costly. Yes, user might make a mistake in his code or get and transfer an incorrect data. Well, an exception will appear. IMHO, this is an admissible risk. WDYT?